visualization/app/codeCharta/ui/copyToClipboardButton/copyToClipboard.service.spec.ts

```diff
@@ -0,0 +1,26 @@
+import { TestBed } from "@angular/core/testing"
+import { CopyToClipboardService } from "./copyToClipboard.service"
+import { buildTextOfFiles } from "./util/clipboardString"
+import { getFilenamesWithHighestMetrics } from "./util/getFilenamesWithHighestMetrics"
+let service: CopyToClipboardService
+jest.mock("./util/clipboardString", () => {
+	return { buildTextOfFiles: jest.fn().mockReturnValue("Magic Monday") }
+})
+jest.mock("./util/getFilenamesWithHighestMetrics", () => {
+	return { getFilenamesWithHighestMetrics: jest.fn() }
+})
+describe("copyToClipboardService", () => {
+	beforeEach(() => {
+		TestBed.configureTestingModule({ providers: [CopyToClipboardService] })
+		service = TestBed.inject(CopyToClipboardService)
+
+		jest.clearAllMocks()
+	})
+	it("should call text function and mapNode function on getClipboardText", () => {
```

Comment:
I also agree with this.

Reply:
For me, it sounds like both of you are fine with deleting this test, but you are not against keeping this test, did I get this right?

If we do not apply some kind of "convert data" and "build text from it", then there is certainly a bug or we are refactoring the code.

As a TDD user, I would benefit from these tests if I ever need to refactor this. I might also benefit from these tests as soon as we advance with the follow-up tickets.

As a non-TDD user, the only case I see where you would get bothered by this test is the case where we start to construct the string directly from the codeMapNode.
And honestly, this does not give us a relevant performance advantage, but it would certainly make the code less readable. And based on your previous opinions, you would advice against this anyway.

So from my point of view, I only see potential benefits and no relevant harm.

Reply:
Whatever makes you happy :grin: